src/renderer_webgpu.cpp

```diff
@@ -0,0 +1,4808 @@
+		numBindings++;
+
+		if (m_numUniforms > 1)
+		{
+			bindings[numBindings].binding = 48;
+			bindings[numBindings].visibility = wgpu::ShaderStage::Fragment;
+			bindings[numBindings].type = wgpu::BindingType::UniformBuffer;
+			bindings[numBindings].hasDynamicOffset = true;
+			numBindings++;
+		}
+
+		uint8_t numSamplers = 0;
+
+		for (uint32_t ii = 0; ii < _vsh->m_numSamplers; ++ii)
+		{
+			m_textures[ii] = _vsh->m_textures[ii];
+			m_samplers[ii] = _vsh->m_samplers[ii];
+			bindings[numBindings++] = _vsh->m_textures[ii];
+			bindings[numBindings++] = _vsh->m_samplers[ii];
+		}
+
+		numSamplers += _vsh->m_numSamplers;
+
+		if (NULL != _fsh)
+		{
+			for (uint32_t ii = 0; ii < _fsh->m_numSamplers; ++ii)
+			{
+				m_textures[numSamplers + ii] = _fsh->m_textures[ii];
+				m_samplers[numSamplers + ii] = _fsh->m_samplers[ii];
+				bindings[numBindings++] = _fsh->m_textures[ii];
+				bindings[numBindings++] = _fsh->m_samplers[ii];
+			}
+
+			numSamplers += _fsh->m_numSamplers;
+		}
+
+		for (uint8_t stage = 0; stage < BGFX_CONFIG_MAX_TEXTURE_SAMPLERS; ++stage)
+		{
+			if (isValid(m_vsh->m_bindInfo[stage].m_uniform))
+			{
+				m_bindInfo[stage] = m_vsh->m_bindInfo[stage];
+			}
+			else if (NULL != m_fsh && isValid(m_fsh->m_bindInfo[stage].m_uniform))
+			{
+				m_bindInfo[stage] = m_fsh->m_bindInfo[stage];
+				m_bindInfo[stage].m_index += _vsh->m_numSamplers;
+			}
+		}
+
+		m_numSamplers = numSamplers;
+
+		for (uint32_t ii = 0; ii < _vsh->m_numBuffers; ++ii)
+		{
+			m_buffers[ii] = _vsh->m_buffers[ii];
+			bindings[numBindings++] = _vsh->m_buffers[ii];
+		}
+
+		m_numBuffers = _vsh->m_numBuffers;
+
+		BX_CHECK(m_numUniforms + m_numSamplers * 2 + m_numBuffers == numBindings, "");
+
+		wgpu::BindGroupLayoutDescriptor bindGroupDesc;
+		bindGroupDesc.entryCount = numBindings;
+		bindGroupDesc.entries = bindings;
+		m_bindGroupLayout = s_renderWgpu->m_device.CreateBindGroupLayout(&bindGroupDesc);
+
+		bx::HashMurmur2A murmur;
+		murmur.begin();
+		murmur.add(m_numUniforms);
+		murmur.add(m_textures, sizeof(wgpu::BindGroupLayoutEntry) * numSamplers);
+		murmur.add(m_samplers, sizeof(wgpu::BindGroupLayoutEntry) * numSamplers);
+		murmur.add(m_buffers,  sizeof(wgpu::BindGroupLayoutEntry) * m_numBuffers);
+		m_bindGroupLayoutHash = murmur.end();
+	}
+
+	void ProgramWgpu::destroy()
+	{
+		m_vsh = NULL;
+		m_fsh = NULL;
+		if ( NULL != m_computePS )
+		{
+			BX_DELETE(g_allocator, m_computePS);
+			m_computePS = NULL;
+		}
+	}
+
+	void BufferWgpu::create(uint32_t _size, void* _data, uint16_t _flags, uint16_t _stride, bool _vertex)
+	{
+		BX_UNUSED(_stride);
+
+		m_size = _size;
+		m_flags = _flags;
+		m_vertex = _vertex;
+
+		const uint32_t paddedSize = bx::strideAlign(_size, 4);
+
+		bool storage = m_flags & BGFX_BUFFER_COMPUTE_READ_WRITE;
+		bool indirect = m_flags & BGFX_BUFFER_DRAW_INDIRECT;
+
+		wgpu::BufferDescriptor desc;
+		desc.size = paddedSize;
+		desc.usage = _vertex ? wgpu::BufferUsage::Vertex : wgpu::BufferUsage::Index;
```

Comment:
The thing is, in current bgfx API you either create vertex or index buffers and no other kind.
All non-index buffers are setup as vertex buffers, with a vertex layout, even if you mean to use them as raw buffers. It would probably make sense, to optimize usage, but it would involve changing bgfx to a deeper level, so not possible ATM I believe (maybe @bkaradzic will correct me)
The last case is when the renderer creates buffer for internal use (e.g staging), but then it does not create them through this function